test_util/test_upgrade_vpc.py

```diff
@@ -38,30 +38,67 @@
     This value will be used when upgrading the cluster.
 
 """
-import collections
-import datetime
 import logging
 import os
+import pprint
 import random
 import string
 import sys
 import uuid
+from contextlib import contextmanager
+
+import retrying
 
 import test_util.aws
 import test_util.cluster
 from pkgpanda.util import load_string
 from test_util.cluster_api import ClusterApi
-from test_util.helpers import CI_AUTH_JSON, DcosUser
-from test_util.marathon import TEST_APP_NAME_FMT
+from test_util.helpers import CI_AUTH_JSON, DcosUser, marathon_app_id_to_mesos_dns_name
 
 
 logging.basicConfig(format='[%(asctime)s|%(name)s|%(levelname)s]: %(message)s', level=logging.DEBUG)
 log = logging.getLogger(__name__)
 
 
-def get_test_app():
-    app = {
-        "id": TEST_APP_NAME_FMT.format(uuid.uuid4().hex),
+TEST_APP_NAME_FMT = 'upgrade-{}'
+
+
+@contextmanager
+def cluster_workload(cluster_api):
+    """Start a cluster workload on entry and verify its health on exit."""
+
+    def _app_tasks(app_id):
```

Comment:
[minor]

These inner functions `_app_tasks` and others are private, namespaced within cluster_workload and cannot be easily accessed from outside. So, they need not be ` _underscore_prefixed`. I think, it is just a habitual thing that we are doing it here.